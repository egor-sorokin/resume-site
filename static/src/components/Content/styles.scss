@import '../../styles/base/_vars.scss';

.project {
  display: flex;
  flex-direction: row;

  width: 100%;
  height: 100%;
}

.col {
}

.col--right {
  width: 67%;
}

.col--left {
  width: 33%;

  padding-right: 5%;
  padding-top: 60px;
  padding-bottom: 60px;

  align-self: center;
}

.image {
  height: 100%;
  background-color: $white;
}

.list {
  margin-top: 10px;
  width: 100%;
}

.list__item {
  overflow: hidden;
}

.list__item:nth-of-type(n+1) {
  margin-top: 20px;
}

.item__plain-text {
  line-height: 1.3;
}

.grid {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.grid__item {
  min-width: 33%;
  min-height: 265px;
  height: 100%;

  padding: 20px 40px;

  background-color: $dune;
  box-shadow: -4px -4px 12px 1px black;
}
