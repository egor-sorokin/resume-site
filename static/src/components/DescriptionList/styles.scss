.description-list {
  margin-top: 10px;
  width: 100%;

  .item {
    overflow: hidden;
  }

  .item:nth-of-type(n+1) {
    margin-top: 20px;
  }

  .item__plain-text {
    line-height: 1.3;
  }
}

