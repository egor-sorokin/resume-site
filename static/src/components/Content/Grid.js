import React from 'react';
import './styles.scss';


const Grid = ({ data }) => {
  const gridComponent = data.map((item) =>
    <li  key={item.id} className="item">
      <a href={item.url}></a>
    </li>
  );

  return (
    <ul>
      {gridComponent}
    </ul>
  );
};


export default Grid;