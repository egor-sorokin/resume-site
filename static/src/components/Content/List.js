import React from 'react';
import './styles.scss';


const List = ({ listData }) => {
  const listComponent = listData.map(item => (
    <li key={item.label} className="list__item item">
      <h6 className="item__title font-f-sec font-s-10 font-w-b text-t-u">
        {item.label}
      </h6>
      {item.url ? (
        <div className="item__link">
          <div className="link-tr">
            <span className="link-tr__placeholder font-s-20 font-w-m text-c-l-dune">{item.text}</span>
            <div className="link-tr__mask">
              <a
                href={item.url}
                target="_blank"
                className="link-tr__text font-s-20 font-w-m text-c-l-dune"
              >
                {item.text}
              </a>
            </div>
          </div>
        </div>
      ) : (
        <p className="item__plain-text font-s-20 font-w-m text-c-l-dune">
          {item.text}
        </p>
      )}
    </li>
  ));

  return (
    <ul className="list">
      {listComponent}
    </ul>
  );
};

export default List;
