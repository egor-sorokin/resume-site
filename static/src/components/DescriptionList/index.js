import React from 'react';
import PropTypes from 'prop-types';
import LinkMasked from '../LinkMasked/index';
import './styles.scss';


const DescriptionList = ({listData}) => {
  const cssClassesLinkMasked = "link-masked--white font-s-20 text-c-l-dune";
  const component = listData.map((item, i) => (
    <li
      key={i+1}
      className="description-list__item item">
      <h6 className="item__title font-s-12-secondary">
        {item.label}
      </h6>
      {item.url ? (
        <div className="item__link">
          <LinkMasked
            linkData={item}
            cssClasses={cssClassesLinkMasked}
            target="_blank"
          ></LinkMasked>
        </div>
      ) : (
        <p className="item__plain-text font-s-20 text-c-l-dune">
          {item.text}
        </p>
      )}
    </li>
  ));

  return (
    <ul className="description-list">
      {component}
    </ul>
  );
};

DescriptionList.propTypes = {
  listData: PropTypes.instanceOf(Array),
};

DescriptionList.defaultProps = {
  listData: [],
};

export default DescriptionList;
