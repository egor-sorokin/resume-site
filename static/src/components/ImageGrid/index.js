import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';


const ImageGrid = ({gridData}) => {
  const component = gridData.map(item => (
    <li
      key={item.id}
      className="image-grid__item item text-c-mercury">
      <a
        href={item.url}
        target="_blank"
        className="item__image image"
        style={{ backgroundImage: `url(${item.image})` }}
      />
    </li>
  ));

  return (
    <ul className="image-grid">
      {component}
    </ul>
  );
};

ImageGrid.propTypes = {
  gridData: PropTypes.instanceOf(Array),
};

ImageGrid.defaultProps = {
  gridData: [],
};

export default ImageGrid;
