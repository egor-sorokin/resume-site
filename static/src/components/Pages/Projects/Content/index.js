import React from 'react';
import Grid from './Grid'
import './styles.scss';


const Content = ({ data }) => (
  <section>
    <h1 class="title">{data.project}</h1>
    <ContentList></ContentList>
    {data.imgUrl ? (
        <div className="content content--right">
          <div className="image" style={{ backgroundImage: `url(${data.imgUrl})` }} />
        </div>
      ) : (
        <Grid></Grid>
      )}
  </section>
);

export default Content;
