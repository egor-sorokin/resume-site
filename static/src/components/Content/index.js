import React from 'react';
import PropTypes from 'prop-types';

import DescriptionList from '../DescriptionList/index';
import ImageGrid from '../ImageGrid/index';
import './styles.scss';


const Content = ({
  name, info, image, experiments,
}) => (
  <section className="project">
    <div className="col col--left">
      <h1 className="font-s-40">
        {name}
      </h1>
      <DescriptionList
        listData={info}
      />
    </div>
    <div className="col col--right">
      {experiments.length === 0 ? (
        <img
          className="image"
          title={image.alt}
          alt={image.alt}
          src={image.src}
        />
      ) : (
        <ImageGrid
          gridData={experiments}
        />
      )}
    </div>
  </section>
);

Content.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.shape({
    alt: PropTypes.string,
    src: PropTypes.string,
  }),
  info: PropTypes.instanceOf(Array),
  experiments: PropTypes.instanceOf(Array),
};

Content.defaultProps = {
  image: {
    alt: '',
    src: '',
  },
  info: [],
  experiments: [],
};

export default Content;
