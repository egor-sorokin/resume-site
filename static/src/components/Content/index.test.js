// eslint-disable-next-line
import {shallow, render, mount} from 'enzyme';
import React from 'react';
import Content from './index';
import ImageGrid from '../ImageGrid/index';
import DescriptionList from '../DescriptionList/index';


describe('<Content />', () => {
  let component;
  const experiments = [
    {
      id: 1,
      label: 'title',
      url: '',
      image: 'assets/exp_name.jpg'
    }
  ];
  const image = {
    alt: 'testalt',
    src: 'testsrc'
  };
  const name = 'Name test';
  const info = [
    {
      label: 'Label test',
      text: 'Text test',
      url: ''
    }
  ];


  describe('base tests', () => {
    beforeEach(() => {
      component = shallow(<Content />);
    });


    it('renders without crashing', () => {
      expect(component).toBeDefined();
    });


    it('matches snapshot', () => {
      expect(component).toMatchSnapshot();
    });


    it('renders 2 components', () => {
      expect(component.children()).toHaveLength(2);
    });
  });


  describe('test setting props', () => {
    beforeEach(() => {
      component = mount(<Content />);
    });


    it('set props correctly', () => {
      component.setProps({name: name, info: info, image: image, experiments: experiments});

      expect('name' in component.props()).toEqual(true);
      expect('info' in component.props()).toEqual(true);
      expect('image' in component.props()).toEqual(true);
      expect('experiments' in component.props()).toEqual(true);
    });
  });


  describe('test props', () => {
    it('renders descriptionList', () => {
      component = shallow(
        <Content info={info}/>
      );

      expect(component.contains(<DescriptionList listData={info}/>)).toBeTruthy();
    });


    it('renders only imageGrid', () => {
      component = shallow(
        <Content experiments={experiments}/>
      );

      expect(component.contains(<img className="image" title={image.alt} alt={image.alt}
                                     src={image.src}/>)).toBeFalsy();
      expect(component.contains(<ImageGrid gridData={experiments}/>)).toBeTruthy();
    });


    it('renders only image', () => {
      component = shallow(
        <Content image={image}/>
      );

      expect(component.contains(<img className="image" title={image.alt} alt={image.alt}
                                     src={image.src}/>)).toBeTruthy();
      expect(component.contains(<ImageGrid gridData={experiments}/>)).toBeFalsy();
    });
  });
});
