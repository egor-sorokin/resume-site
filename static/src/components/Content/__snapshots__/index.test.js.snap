// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> base tests matches snapshot 1`] = `
<section
  className="project"
>
  <div
    className="col col--left"
  >
    <h1
      className="font-s-40"
    />
    <DescriptionList
      listData={Array []}
    />
  </div>
  <div
    className="col col--right"
  >
    <img
      alt=""
      className="image"
      src=""
      title=""
    />
  </div>
</section>
`;
